src/components/AgreementLineSections/FormEresource.js

```diff
@@ -90,10 +90,15 @@ const FormEresource = ({
               addButtonLabel={<FormattedMessage id="ui-agreements.agreementLine.linkSelectedEresource" />}
               basket={basket}
               component={BasketSelector}
-              error={meta.touched && meta.error}
+              error={(meta.touched && meta.error) || meta.data.error}
               label={<FormattedMessage id="ui-agreements.eresource" />}
               name={input.name}
               onAdd={resource => {
+                if (isEmpty(resource.id)) {
+                  setFieldData('linkedResource', { error: <FormattedMessage id="ui-agreements.basketSelector.selectResourceMessage" /> });
+                  input.onChange();
```

Comment:
You could still pass in the resource to `onChange` right? Is that causing an issue?

Reply:
yes, that's causing an error. I thought I fixed it with omitting the resource, but that's not the case. 
When I click the "Link selected e-resource" button without having selected an item, the "Save & close" button becomes active. If I then click "Save & close" I get an error.
<img width="455" alt="Bildschirmfoto 2021-07-13 um 12 15 33" src="https://user-images.githubusercontent.com/42971992/125435044-6905dbda-5cea-43d2-8eb9-f627f735673e.png">

Reply:
So this error ^^^ is still unresolved. Any ideas?

Reply:
The `onAdd` callback should give you an empty object with `{id: ''}`. Calling `input.onChange(resource}` sets the value of the `Field` to be `{id: ''}`. You could then check if there is an `id` on the value within the `validate` function and if there is an id key and its value is an empty string you can return the custom error you want to.

Reply:
The save and close isnt disable in this case here because we are setting the error at `meta.data.error` rather than `meta.error`. Save and close is disabled only if the `meta.error` value is set. So, current solution of `setFieldData` shouldn't be used in this case.